### PR TITLE
fix(lambda-nodejs): esbuild define parameters are incorrectly encoded

### LINKED ISSUE #13842: (aws-lambda-nodejs): "define" bundling option is not escaped
"define" bundling option appears to be insufficiently escaped

### Reproduction Steps
```typescript
const function = new NodejsFunction(this, 'Function', {
    entry: './app/index.ts',
    bundling: {
        define: {
            'process.env.NODE_ENV': JSON.stringify('production'),
        },
    },
})
```

### What did you expect to happen?

Expected would be an option `--define:process.env.NODE_ENV=\"production\"` passed to esbuild

### What actually happened?

esbuild emits a warning:
```
 > warning: "process.env.NODE_ENV" is defined as an identifier instead of a string (surround "production" with double quotes to get a string)
```

suggesting that the quotes, added by `JSON.stringify` were removed, I assume, by the shell executing the esbuild command

### Environment

  - **CDK CLI Version  :** 1.95.1 (build ed2bbe6)
  - **Node.js Version:** v15.12.0
  - **OS               :** Linux 5.11.9, bash 5.1.4
  - **Language (Version):** TypeScript 4.2.3

### Other

Manually escaping all double quotes with `JSON.stringify('production').replace(/"/g, '\\"')` is a workaround I use for now

I also expect that it would be easier to bundle the esbuild's js-api, since then there is less dealing with string escapes. (https://esbuild.github.io/getting-started/#build-scripts)

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by skyrpex**
Fixes #13842

### CODE REVIEW

**Comment by skyrpex** [comment]
There's still invalid edge cases.

**Comment by skyrpex** [comment]
Alright, in order to fix the esbuild define instructions for the CLI, I had to JSON.stringify all the values. Now they work as expected.

Added a snapshot test that calls the esbuild CLI to ensure that this doesn't break again.

File: packages/@aws-cdk/aws-lambda-nodejs/test/bundling.test.ts
Comment (jogold) [inline packages/@aws-cdk/aws-lambda-nodejs/test/bundling.test.ts:200]:
```suggestion
  expect(bundleProcess.stdout.toString()).toMatchSnapshot();
```

File: packages/@aws-cdk/aws-lambda-nodejs/test/bundling.test.ts
Comment (jogold) [inline packages/@aws-cdk/aws-lambda-nodejs/test/bundling.test.ts:198]: Do you think we really need to run `esbuild` here in the test? Isn't it enough to ensure that the right command line args are passed?

**Review by jogold — COMMENTED** [review]
Thanks @skyrpex!

Some minor comments.

File: packages/@aws-cdk/aws-lambda-nodejs/test/bundling.test.ts
Comment (skyrpex) [reply packages/@aws-cdk/aws-lambda-nodejs/test/bundling.test.ts:198]: Well, I got it right this time and reverse-engineered the define parameters' expectation based on that (it's a hell of escape slashes). It should be stable now and the test is not exactly necessary, but it's proof that it works now.

Let me know if you want me to remove it 👍🏻

Comment (eladb) [reply packages/@aws-cdk/aws-lambda-nodejs/test/bundling.test.ts:198]: I like that this is doing an end to end test

**Review by eladb — APPROVED** [review]

**Comment by eladb** [comment]
@jogold waiting for your approval and I'll remove the do-not-merge label

**Review by jogold — APPROVED** [review]

**Pull request merged**
